pandas/core/indexes/multi.py

```diff
@@ -2203,19 +2203,23 @@ def append(self, other):
         if all(
             (isinstance(o, MultiIndex) and o.nlevels >= self.nlevels) for o in other
         ):
-            arrays = []
+            arrays, names = [], []
             for i in range(self.nlevels):
                 label = self._get_level_values(i)
                 appended = [o._get_level_values(i) for o in other]
                 arrays.append(label.append(appended))
-            return MultiIndex.from_arrays(arrays, names=self.names)
+                level_names = {label.name}.union({x.name for x in appended})
```

Comment:
Could simplify to:

```
single_label_name = all(label.name == x.name for x in appended)
names.append(label.name if single_label_name else None)
```

so the name checking could short circuit if possible.

Reply:
makes sense, thx